test/tests/functional/pbs_qstat_validate_attr.py

```diff
@@ -0,0 +1,117 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestQstatAttrValidate(TestFunctional):
+
+    def validate_el(self, attr_dict, user):
+        qstat_o = {}
+        qstat_o['server_state'] = 'Active'
+        qstat_o['server_host'] = self.server.hostname
+        qstat_o['scheduling'] = 'True'
+        qstat_o['total_jobs'] = '0'
+        qstat_o['state_count'] = ['Transit', 'Queued',
+                                  'Held', 'Waiting', 'Running',
+                                  'Exiting', 'Begun']
+        qstat_o['default_queue'] = 'workq'
+        qstat_o['log_events'] = '511'
+        qstat_o['mail_from'] = 'adm'
+        qstat_o['query_other_jobs'] = 'True'
+        qstat_o['resources_default.ncpus'] = '1'
+        qstat_o['default_chunk.ncpus'] = '1'
+        qstat_o['scheduler_iteration'] = '600'
+        qstat_o['resv_enable'] = 'True'
+        qstat_o['node_fail_requeue'] = '310'
+        qstat_o['max_array_size'] = '10000'
+        qstat_o['pbs_license_linger_time'] = '3600'
+        qstat_o['pbs_version'] = self.server.pbs_version
+        qstat_o['eligible_time_enable'] = 'False'
+        qstat_o['max_concurrent_provision'] = '5'
+
+        if user == 'root':
+            qstat_o['power_provisioning'] = 'False'
+
+        platform = self.du.get_platform()
+        if platform == 'cray' or platform == 'craysim':
+            qstat_o['restrict_res_to_release_on_suspend'] = 'ncpus'
+
+        for each in attr_dict:
+            if each in qstat_o:
+                if "count" in each:
+                    out1 = []
+                    out = attr_dict[each].split()
+                    for k in out:
+                        out1.append(k.split(':')[0])
+                    self.assertListEqual(qstat_o[each], out1)
+                else:
+                    self.assertEqual(qstat_o[each], attr_dict[each])
+
+    @staticmethod
+    def filter_el(e):
+        for each in e:
+            if "=" in each:
+                return each
+
+    def test_qstat_Bf_as_user(self):
+        attr_dict = {}
+        command = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                               'bin', 'qstat -Bf')
+        ret = self.du.run_cmd(self.server.hostname, command)
```

Comment:
any reason why are you not using server.status

Reply:
thank you for this suggestion. Helped in reducing lines of code.